Failing tests in time package

## Description

There are failing tests on the main branch when running them locally

---

## Steps to Reproduce (for bugs)

Running ``make test`` on the main branch.

---

## Expected Behavior

All tests should pass

---

## Actual Behavior

```bash
bash ./scripts/test.sh
ok  	github.com/kashifkhan0771/utils/boolean	(cached)
ok  	github.com/kashifkhan0771/utils/ctxutils	(cached)
ok  	github.com/kashifkhan0771/utils/errutils	(cached)
ok  	github.com/kashifkhan0771/utils/fake	(cached)
ok  	github.com/kashifkhan0771/utils/maps	(cached)
ok  	github.com/kashifkhan0771/utils/math	(cached)
ok  	github.com/kashifkhan0771/utils/pointers	(cached)
ok  	github.com/kashifkhan0771/utils/rand	(cached)
ok  	github.com/kashifkhan0771/utils/slice	(cached)
ok  	github.com/kashifkhan0771/utils/strings	(cached)
ok  	github.com/kashifkhan0771/utils/structs	(cached)
ok  	github.com/kashifkhan0771/utils/templates	(cached)
--- FAIL: TestCalculateAge (0.00s)
    --- FAIL: TestCalculateAge/Leap_year_adjustment,_after_birthday (0.00s)
        time_test.go:498: CalculateAge() = 24, want 25
FAIL
FAIL	github.com/kashifkhan0771/utils/time	0.003s
ok  	github.com/kashifkhan0771/utils/url	(cached)
FAIL
make: *** [Makefile:3: test] Error 1
```

---

## Environment

- **Go Version:** 1.23.3
- **Operating System:** EndeavourOS x86_64
- **Version:** 1.0.0

---

## Additional Context

The tests are also failing in my [PR](https://github.com/kashifkhan0771/utils/pull/85)

--- a/time/time_test.go
+++ b/time/time_test.go
@@ -6,6 +6,9 @@ import (
 	"time"
 )
 
+// time.Now mock
+var now = time.Date(2025, 1, 10, 0, 0, 0, 0, time.UTC)
+
 func TestStartOfDay(t *testing.T) {
 	type args struct {
 		input time.Time
@@ -346,7 +349,7 @@ func TestConvertToTimeZone(t *testing.T) {
 				location: "Local",
 			},
 			wantErr: false,
-			want:    time.Date(2023, 12, 25, 15, 0, 0, 0, time.UTC).In(time.Now().Location()),
+			want:    time.Date(2023, 12, 25, 15, 0, 0, 0, time.UTC).In(now.Location()),
 		},
 		{
 			name: "Convert PST to IST",
@@ -445,7 +448,7 @@ func TestCalculateAge(t *testing.T) {
 		{
 			name: "Birthday today, age remains the same",
 			args: args{
-				birthDate: time.Date(time.Now().Year()-30, time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.UTC),
+				birthDate: time.Date(now.Year()-30, now.Month(), now.Day(), 0, 0, 0, 0, time.UTC),
 			},
 			want: 30,
 		},
@@ -475,7 +478,13 @@ func TestCalculateAge(t *testing.T) {
 			args: args{
 				birthDate: time.Date(2000, 2, 29, 0, 0, 0, 0, time.UTC),
 			},
-			want: time.Now().Year() - 2000,
+			want: func() int {
+				leapYearAge := now.Year() - 2000
+				if now.Before(time.Date(now.Year(), 2, 29, 0, 0, 0, 0, time.UTC)) {
+					leapYearAge--
+				}
+				return leapYearAge
+			}(),
 		},
 		{
 			name: "Future date, invalid age",

